features/yield/FarmingCalculator.tsx

```diff
@@ -137,13 +137,21 @@ const FarmingCalculator = () => {
         [key: string]: number;
       }
       const usdRewards: rewardTokenMap = {};
-      const tokenPrices: rewardTokenMap = {};
+
+      // Fetch token prices if not set already.
+      let tokenPrices: rewardTokenMap = rewardTokenPrices;
       for (let rewardObj of rewardToken) {
-        const _tokenPrice = await rewardObj.getPrice();
-        tokenPrices[rewardObj.token] = _tokenPrice;
+        let _tokenPrice;
```

Comment:
This code was unintentionally calling the coinGeckoApi frequently to load new `_tokenPrices` into the `rewardTokenPrices` state var. 

Here, I
- set `tokenPrices` by default to the existing `rewardTokenPrices` struct
- if `tokenPrices[token]` already exists, no need to call the API again
- Otherwise, call the API and load it into the state